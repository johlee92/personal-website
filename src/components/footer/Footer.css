.footer-container {
    flex-shrink: 0;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    padding: 10px 0;
    align-items: center;
    border-top: darkgray 2px solid;
}

.footer-container > p {
    margin: 0;
    font-size: 18px;
    font-weight: bold;
}

.footer-image {
    width: 25px;
    height: 25px;
}