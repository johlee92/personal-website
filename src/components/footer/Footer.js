import React, { Component } from 'react';
import './Footer.css';
import { Link } from 'react-router-dom';

class Footer extends Component {
    render() {
        return (
            <div className="footer-container">
                <span></span>
                <span></span>
                <span>
                    <Link to='/contact' style={{ textDecoration: 'none' }}>Contact</Link>
                </span>
            </div>
        )
    }
}

export default Footer;