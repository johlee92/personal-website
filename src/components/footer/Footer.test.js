import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Footer from './Footer';

describe(`Footer component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
                <Footer />
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component as expected', () => {
        const tree = renderer
            .create(
                <Footer />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})
