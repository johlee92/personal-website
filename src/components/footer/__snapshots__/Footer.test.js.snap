// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders the component as expected 1`] = `
<div
  className="footer-container"
>
  <span />
  <span />
  <span>
    <a
      href="/contact"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Contact
    </a>
  </span>
</div>
`;
